test/federation_mgmt_SUITE.erl

```diff
@@ -251,14 +251,15 @@ assert_code(CodeExp, CodeAct, Type, Path, Body) ->
                           path, Path, body, Body})
     end.
 
-decode(?OK, _Headers,  ResBody) -> cleanup(rabbit_json:decode(list_to_binary(ResBody)));
+decode(?OK, _Headers,  ResBody) ->
+    cleanup(rabbit_json:decode(list_to_binary(ResBody), [return_maps]));
 decode(_,    Headers, _ResBody) -> Headers.
 
 cleanup(L) when is_list(L) ->
     [cleanup(I) || I <- L];
-cleanup({struct, I}) ->
-    cleanup(I);
-cleanup({K, V}) when is_binary(K) ->
-    {list_to_atom(binary_to_list(K)), cleanup(V)};
+cleanup(M) when is_map(M) ->
+    maps:fold(fun(K, V, Acc) ->
+        Acc#{binary_to_atom(K, latin1) => cleanup(V)}
```

Comment:
Good point:
I got the same from:
https://github.com/rabbitmq/rabbitmq-shovel-management/blob/master/test/http_SUITE.erl#L262
https://github.com/rabbitmq/rabbitmq-tracing/blob/master/test/rabbit_tracing_SUITE.erl#L229

When we refactor these functions we could change it